model/src/main/java/org/hibernate/performance/search/model/asset/AutomaticIndexingState.java

```diff
@@ -43,98 +46,95 @@ public AutomaticIndexingState(RelationshipSize relationshipSize, int initialInde
 		}
 	}
 
-	public synchronized void start() {
+	public synchronized void startTrial() {
 		if ( started ) {
 			return;
 		}
 		sessionFactory = HibernateORMHelper.buildSessionFactory( additionalProperties );
-		DomainDataFiller domainDataFiller = new DomainDataFiller( sessionFactory, relationshipSize );
-		for ( int i = 0; i < initialIndexSize; i++ ) {
-			domainDataFiller.fillData( i );
-		}
-		indexInsertPartitions = createInsertPartitions();
-		started = true;
+		start();
 	}
 
-	public synchronized void stop() {
+	public synchronized void stopTrial() {
 		if ( !started ) {
 			return;
 		}
 		if ( sessionFactory != null ) {
 			sessionFactory.close();
 		}
-		indexInsertPartitions = null;
-		indexUpdatePartitions = null;
-		indexDeletePartitions = null;
-		started = false;
+		stop();
 	}
 
-	public AutomaticIndexingInsertPartitionState getInsertPartition(int threadNumber) {
-		checkThreadNumber( threadNumber );
-		return indexInsertPartitions.get( threadNumber );
+	public synchronized void startIteration() {
+		if ( started ) {
+			return;
+		}
+		start();
 	}
 
-	public AutomaticIndexingUpdatePartitionState getUpdatePartition(int threadNumber) {
-		checkThreadNumber( threadNumber );
-		synchronized (this) {
-			if ( indexUpdatePartitions == null ) {
-				indexUpdatePartitions = createUpdatePartitions();
+	public synchronized void stopIteration() {
+		if ( !started ) {
+			return;
+		}
+		new DomainDataRemover( sessionFactory ).truncateAll();
```

Comment:
If you're going to do this, you also need to initialize the data when the iteration starts.
Otherwise, the first (warmup) iteration will work on a large database and index, and the next ones will work on an almost-empty database and index.

Another approach would be to keep the data between iterations, but I don't know how feasible that would be. It's probably too complex, so I wouldn't try unless initializing the data is very slow.

Reply:
I think that we already do that if we start an iteration : iteration number > 1. Is that possible?

Reply:
You're right. Somehow I didn't notice that you call `stop` after each iteration, leading to `start` being executed before each iteration, leading to data initialization before each iteration.